### ISSUE TITLE
Discord 0.0.9 fails to launch

### ISSUE THREAD

**Issue by buenchico** [flathub/com.discordapp.Discord#73]
Hi,

discord fails to launch. Running it on console produce:
```
$ flatpak run com.discordapp.Discord                          
Gtk-Message: 18:47:59.958: Failed to load module "canberra-gtk-module"
Discord 0.0.9
Starting app.
Starting updater.
[Modules] Modules initializing
[Modules] Distribution: remote
[Modules] Host updates: enabled
[Modules] Module updates: enabled
[Modules] Module install path: /home/buenchico/.var/app/com.discordapp.Discord/config/discord/0.0.9/modules
[Modules] Module installed file path: /home/buenchico/.var/app/com.discordapp.Discord/config/discord/0.0.9/modules/installed.json
[Modules] Module download path: /home/buenchico/.var/app/com.discordapp.Discord/config/discord/0.0.9/modules/pending
[Modules] No updates to install
[Modules] Checking for host updates.
[Modules] Host is up to date.
[Modules] Checking for module updates at https://discordapp.com/api/modules/stable/versions.json
[Modules] No module updates available.
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 1000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 1994.1441612867388 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 3471.232217695405 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 9709.191002417268 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 12624.199343672812 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 19146.52462664575 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
(node:4) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 resume listeners added. Use emitter.setMaxListeners() to increase limit
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms
[WebContents] retrying load https://discordapp.com/app?_=1563990480860
[WebContents] did-fail-load -202 ERR_CERT_AUTHORITY_INVALID retry in 20000 ms

```

Discord loading screen appears and an empty window is rendered without actually launching the app.

**Comment by refi64** [comment]
Out of curiosity, what is your host distro?

**Comment by buenchico** [comment]
Mageia 7

**Comment by refi64** [comment]
Interesting, the only other report I've seen for this bug was also on Mageia.

**Comment by ruslansin** [comment]
I have the same problem for Arch Linux. It happened about 2 days ago. I tried to delete .var/<discord_app_dir> and reinstall the app, but it didn't help.
[discord_terminal_output.txt](https://github.com/flathub/com.discordapp.Discord/files/4107845/discord_terminal_output.txt)

**Comment by lae** [comment]
Yeah, can confirm this started happening for me on Arch Linux after restarting my client today as well (and tried the same thing troubleshooting steps Ruslan did before I landed on this issue). This doesn't happen with Discord outside of flatpak.

**Comment by Vash63** [comment]
Same problem on Arch here as of 2 days ago

**Comment by flyingpimonster** [comment]
I'm experiencing this on Fedora 32.

**Comment by realnc** [comment]
I've been using discord from flatpak for quite a while now. It was working fine. Today, I'm getting the same error.

I'm on Gentoo Linux. The Gentoo packages that got updated between Discord working and not working are:

sys-apps/shadow-4.8.1 (from 4.8)
app-crypt/p11-kit-0.23.19 (from 0.23.18.1)
dev-libs/nss-3.49.2 (from 3.49.1)
mail-client/thunderbird-bin-68.4.2 (from 68.4.1)

**Comment by realnc** [comment]
OK, I downgraded p11-kit back to 0.23.18.1 and Discord works again.

**Comment by turtle-bazon** [comment]
Also affects Slack and, I think, other applications with access to services.

**Comment by dac73** [comment]
Same problem on Manjaro as of today (probably yesterday's update).
Downgrade fixes the issue.

**Comment by philmmanjaro** [comment]
So something is off with **p11-kit 0.23.19-1**. Did anyone do a **git-bisect** yet?

**Comment by nightuser** [comment]
@philmmanjaro 

API change: https://github.com/p11-glue/p11-kit/commit/1def8077a2bc1fc2a6bd3685a9d94a9a51f40e23

Upstream flatpak issue: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2247

**Comment by Vash63** [comment]
> @philmmanjaro
> 
> API change: [p11-glue/p11-kit@1def807](https://github.com/p11-glue/p11-kit/commit/1def8077a2bc1fc2a6bd3685a9d94a9a51f40e23)
> 
> Upstream flatpak issue: https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/merge_requests/2247
> 
> UPD: It seems that the fix is in the works (the latest update on the issue was less than 30 mins ago). And hopefully, the fix is trivial (just updating p11-kit in the platform).

If the fix is just to update p11-kit in the platform I worry that this means this will happen every time that package is updated on my system... isn't flatpak supposed to prevent compatibility issues, not make them worse?

**Comment by TingPing** [comment]
Flatpak sadly does not maintain p11-kit. I have no idea why they broke but that discussion probably belongs wit them.

**Comment by ebbit1q** [comment]
This issue has been solved for me, flatpak 1.6.1 is now in arch repositories.

**Comment by TingPing** [comment]
It was fixed by a p11-kit update not by flatpak.
